File: android/src/main/java/com/rivereactnative/RNDirection.kt
package com.rivereactnative

import app.rive.runtime.kotlin.core.Direction

enum class RNDirection(private val mValue: String) {
  Backwards("backwards"),
  Auto("auto"),
  Forwards("forwards");

  override fun toString(): String {
    return mValue
  }

  companion object {
    fun mapToRNDirection(direction: String): RNDirection {
      return valueOf(values().first() { it.toString() == direction }.name)
    }

    fun mapToRiveDirection(rnDirection: RNDirection): Direction {
      return when (rnDirection) {
        Backwards -> Direction.BACKWARDS
        Auto -> Direction.AUTO
        Forwards -> Direction.FORWARDS
      }
    }
  }
}

File: ios/RNDirection.swift
enum RNDirection: String {
    case Backwards = "backwards"
    case Auto = "auto"
    case Forwards = "forwards"

    static func mapToRNDirection(value: String) -> RNDirection {
        if let rnEnum = RNDirection(rawValue: value) {
            return rnEnum
        } else {
            // Return a default value instead of crashing
            RCTLogWarn("Unsupported direction type: \(value), defaulting to Auto")
            return .Auto
        }
    }

    static func mapToRiveDirection(rnDirection: RNDirection) -> RiveDirection {
        switch rnDirection {
        case .Backwards:
            return .backwards
        case .Auto:
            return .autoDirection
        case .Forwards:
            return .forwards
        }
    }
}
